Purchases/Public/Purchases.swift

```diff
@@ -175,6 +175,7 @@ public typealias PaymentDiscountBlock = (SKPaymentDiscount?, Error?) -> Void
     /// Current version of the Purchases framework.
     @objc public static var frameworkVersion: String { SystemInfo.frameworkVersion }
 
+    @available(*, deprecated, message: "Configure behavior through the RevenueCat dashboard instead")
```

Comment:
All of these new deprecations are causing build errors in the ObjC API tester. I'm a little confused why this is happening, since `debugLogsEnabled` is called, and builds with just an error. Perhaps there's a difference in treatment between properties and functions? Or did we ever configure something to treat certain deprecation errors as warnings?

Either way, I'd think warnings are preferable unless we want to officially drop support for those methods.

I confirmed that in Swift, these only trigger build warnings...

Reply:
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
    // deprecate debugLogsEnabled call here
#pragma clang diagnostic pop

Reply:
I turned on build warnings as errors in the API tester project. Context:
If we change the nullability of something and we pass in `nil` in objc, it's not a build error- just a warning.
(https://github.com/RevenueCat/purchases-ios/pull/827)

Reply:
> ```
> #pragma clang diagnostic push
> #pragma clang diagnostic ignored "-Wdeprecated-declarations"
>     // deprecate debugLogsEnabled call here
> #pragma clang diagnostic pop
> ```

we don't have that around the current debugLogs call, though, which is what was confusing me...although now this doesn't appear to be triggering a warning either 🤯 :
`debugLogsEnabled = [RCPurchases debugLogsEnabled];`

Reply:
welp, not sure what could've changed but now i'm getting build _warnings_ for the ones i added, and still nothing for `debugLogsEnabled`.

Reply:
Yeah, that makes sense. I think for the tester project that's ok. It would be nice to be able to specify places where we want to emit a warning or an error.